ui-v2/app/controllers/dc/acls/policies/edit.js

```diff
@@ -0,0 +1,38 @@
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { get, set } from '@ember/object';
+export default Controller.extend({
+  builder: service('form'),
+  dom: service('dom'),
+  isScoped: false,
+  setProperties: function(model) {
```

Comment:
> Overriding setProperties like this is bad imo

It's a public method? Could you explain why you think it's bad? Again I'm already doing this in other places in the app (including the old ACLs that are being replaced by this)

I don't think `setupController` fits my needs here, I know of at least one place where I am calling `setProperties` from the route after controller setup, so if I moved it to `setupController` as far as I understand it my extra code then wouldn't be called?

Reply:
Just the principle of least surprise - `setProperties` is usually for setting a hash on an ember object. I found it surprising that suddenly it has side effects and that for things like forms to work you had to call `setProperties`.

Reply:
Ah getcha, yeah I suppose. Here as the forms basically wrap the hash you send in with a `ember-validation-changeset` which is just a proxy object to the hash you've sent.

You don't 'have' to do this in `setProperties`, you could do it anywhere where you have the properties available. I just know everywhere in the app the first time the hash is set is in `setProperties`, then I know that whenever I mutate those I always use `setProperties` to do it.

We can chat more tomorrow, this is one of the things it would be good to chat through with you.